app/controllers/api/avatars.php

```diff
@@ -366,12 +369,17 @@
 
         $download = ($download === '1' || $download === 'true' || $download === 1 || $download === true);
 
-        $renderer = new ImageRenderer(
+       /*  $renderer = new ImageRenderer(
             new RendererStyle($size, $margin),
             new ImagickImageBackEnd('png', 100)
-        );
+        ); */
+        $qropts = new QROptions([
+            'quietzone'         => $size
+        ]);
+        $qrcode = new QRCode($qropts);
+        $qrcode->render($text);
```

Comment:
This line is not doing anything: `$qrcode->render($text);` just return the result output to no var. What you should do is use the `->send` method in the bottom of the route to attach the image data to our response payload. Example:

```php   
        $response
            ->addHeader('Expires', \date('D, d M Y H:i:s', \time() + (60 * 60 * 24 * 45)).' GMT') // 45 days cache
            ->setContentType('image/png')
            ->send($qrcode->render($text))
        ;
```